src/plugins/intel_gpu/tests/test_cases/matrix_nms_gpu_test.cpp

```diff
@@ -0,0 +1,674 @@
+
+        network network(engine, topology);
+        network.set_input_data("boxes", boxes);
+        network.set_input_data("scores", scores);
+
+        auto outputs = network.execute();
+
+        auto output = outputs.at("matrix_nms").get_memory();
+        cldnn::mem_lock<T> output_ptr(output, get_test_stream());
+
+        cldnn::mem_lock<int> selected_boxes_ptr(selected_boxes, get_test_stream());
+        cldnn::mem_lock<int> valid_outputs_ptr(valid_outputs, get_test_stream());
+
+        const auto expected_output = convert<T>(test_inputs.expected_output);
+        ASSERT_EQ(expected_output.size(), output_ptr.size());
+        for (size_t i = 0; i < expected_output.size(); ++i) {
+            EXPECT_NEAR(expected_output[i], output_ptr[i], THRESHOLD);
+        }
+
+        ASSERT_EQ(test_inputs.expected_selected_boxes.size(), selected_boxes_ptr.size());
+        for (size_t i = 0; i < test_inputs.expected_selected_boxes.size(); ++i) {
+            EXPECT_EQ(test_inputs.expected_selected_boxes[i], selected_boxes_ptr[i]);
+        }
+
+        ASSERT_EQ(test_inputs.expected_valid_outputs.size(), valid_outputs_ptr.size());
+        for (size_t i = 0; i < test_inputs.expected_valid_outputs.size(); ++i) {
+            EXPECT_EQ(test_inputs.expected_valid_outputs[i], valid_outputs_ptr[i]);
+        }
+    }
+
+    static std::string PrintToStringParamName(const testing::TestParamInfo<matrix_nms_test_params>& info) {
+        auto& test_inputs = std::get<0>(info.param);
+        std::ostringstream result;
+
+        auto sort_res_type_str =
+            test_inputs.sort_result_type == matrix_nms::SortResultType::score
+                ? "score"
+                : test_inputs.sort_result_type == matrix_nms::SortResultType::class_id ? "class_id" : "none";
+        auto decay_function_str =
+            test_inputs.decay_function == matrix_nms::DecayFunction::linear
+                ? "linear"
+                : test_inputs.decay_function == matrix_nms::DecayFunction::gaussian ? "gaussian" : "none";
+
+        result << "SortResultAcrossBatch=" << bool_to_str(test_inputs.sort_result_across_batch) << "_";
+        result << "ScoreThreshold=" << test_inputs.score_threshold << "_";
+        result << "NmsTopK=" << test_inputs.nms_top_k << "_";
+        result << "KeepTopK=" << test_inputs.keep_top_k << "_";
+        result << "BackgroundClass=" << test_inputs.background_class << "_";
+        result << "GaussianSigma=" << test_inputs.gaussian_sigma << "_";
+        result << "PostThreshold=" << test_inputs.post_threshold << "_";
+        result << "Normalized=" << bool_to_str(test_inputs.normalized) << "_";
+        result << "SortResultType=" << sort_res_type_str << "_";
+        result << "DecayFunction=" << decay_function_str << "_";
+        result << "Format=" << fmt_to_str(std::get<1>(info.param));
+
+        if (!test_inputs.test_name.empty()) {
+            result << "_" << test_inputs.test_name;
```

Comment:
Do we print "test_inputs.test_name" two times here for some reason?

Reply:
done